File: 38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        '''
        need a helper and just start from beginning and go through to the end
        '''
        res = '1'
        for i in range(n-1):
            res = self.next(res)
        return res
        
    def next(self, s):
        res = ''
        count = 0
        for i, char in enumerate(s):
            if i == 0:
                count += 1
            elif char == s[i-1]:
                count += 1
            else:
                res += str(count)
                res += str(s[i-1])
                count = 1
        res += str(count)
        res += str(s[-1])
        return res

File: 38. Count and Say.java
class Solution {
    public String countAndSay(int n) {
        String ans = "1";
        if (n==1) {
            return ans;
        }
        int count = 0;
        char c = '1';
        for (int i=1; i<n; i++) {
            StringBuilder sb = new StringBuilder();
            count = 0;
            c = '1';
            for (int j=0; j<ans.length(); j++) {
                if (c == ans.charAt(j)) {
                    count++;
                } else {
                    if (count == 0) {
                        c = ans.charAt(j);
                        count = 1;
                    } else {
                        sb.append(Integer.toString(count));
                        sb.append(c);
                        c = ans.charAt(j);
                        count = 1;
                    }
                }
            }
            sb.append(Integer.toString(count));
            sb.append(c);
            ans = sb.toString();
        }
        return ans;
    }
}